v2/tools/generator/internal/config/type_transformer.go

```diff
@@ -55,37 +56,163 @@ type TypeTransformer struct {
 	matchedProperties map[astmodel.TypeName]string
 }
 
-func produceTargetType(
-	target TransformTarget,
+func (target *TransformTarget) AppliesToType(t astmodel.Type) bool {
+	if target.appliesCache == nil {
+		target.appliesCache = make(map[astmodel.Type]bool)
+	}
+
+	if result, ok := target.appliesCache[t]; ok {
+		return result
+	}
+
+	result := target.appliesToType(t)
+	target.appliesCache[t] = result
+	return result
+}
+
+func (target *TransformTarget) appliesToType(t astmodel.Type) bool {
+	if target == nil {
```

Comment:
Does this ever actually happen? Do we actually call `appliesToType` on a `nil` `target`? Guarding against this is sorta weird but it'd be weird if this ever happened too?

Reply:
It seemed like an easy way to remove boilerplate from calling methods. A **nil** value can happen if a matcher is never populated (say, if it is not used in the YAML).